asciimatics/widgets/textbox.py

```diff
@@ -283,6 +301,35 @@ def _reflowed_text(self):
 
         return self._reflowed_text_cache
 
+    @property
+    def hide_cursor(self):
+        """
+        Set to True to stop the cursor from showing. If the cursor is hidden,
+        scrolling will move the entire TextBox instead of just the cursor.
+
+        Defaults to False.
+        """
+        return self._hide_cursor
+
+    @hide_cursor.setter
+    def hide_cursor(self, new_value):
+        self._hide_cursor = new_value
+
+    @property
+    def auto_scroll(self):
+        """
+        When set to True the TextBox will scroll to the bottom when created or
+        next text is added. When set to False, the current scroll position
+        will remain even if the contents are changed.
+
+        Defaults to True.
+        """
+        return self._auto_scroll
+
```

Comment:
Any chance you could add a test or two for these new properties - e.g. to check defaults and verify auto-scroll works as intended when switched off?

Reply:
Hey Peter,

A new client has popped up for me, so I'm not sure when I'll get around to this. I can commit what I have, without the test so it isn't sitting and then add the test to the other PR for whenever it gets done, or I can sit on this until I get time. Whichever is best for you.

Reply:
Thanks for letting me know.  Why don't you push what you have now and if I get time before you, I can create the tests before I merge?

Reply:
Ok, I merged down from master and it looks like everything is passing. I've added this to my to-do, we'll see who gets to it first.